src/Uno.UI/Microsoft/UI/Xaml/Controls/RatingControl/RatingControl.mux.cs

```diff
@@ -398,9 +399,10 @@ private void ApplyScaleExpressionAnimation(UIElement uiElement, int starIndex)
 		uiElementVisual.StartAnimation("Scale.X", ea);
 		uiElementVisual.StartAnimation("Scale.Y", ea);
 
-		// Star size = 16. 0.5 and 0.8 are just arbitrary center point chosen in design spec
+		// Star size = 16. 0.5 is just arbitrary center point chosen in design spec
 		// 32 = star size * 2 because of the rendering at double size we do
-		uiElementVisual.CenterPoint = new Vector3(c_defaultRatingFontSizeForRendering * c_horizontalScaleAnimationCenterPoint, c_defaultRatingFontSizeForRendering * c_verticalScaleAnimationCenterPoint, 0.0f);
+		var scaledStarSize = c_defaultRatingFontSizeForRendering * c_scaleAnimationCenterPoint;
+		uiElementVisual.AnchorPoint = new Vector2(scaledStarSize);
```

Comment:
This code originated from WinUI.

The first step I would do is to ensure the source is synchronized with latest from WinUI. If it still doesn't work, it's likely there is a bug somewhere else.

Reply:
Indeed, the code has diverged from WinUI since it was ported, it warrants overhaul first and then we should validate what is potentially still missing in terms of Composition APIs in Skia to make it work as is